lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java

```diff
@@ -2569,6 +2576,111 @@ void should_ApplyFilter_When_QueryingForNotWithin() {
       }
     }
 
+    @Nested
+    @TestInstance(Lifecycle.PER_CLASS)
+    class WithoutAttachment {
+      WorkbasketSummary wb;
+      TaskSummary taskSummary1;
+      TaskSummary taskSummary2;
+      TaskSummary taskSummary3;
+      TaskSummary taskSummary4;
+
+      @WithAccessId(user = "user-1-1")
+      @BeforeAll
+      void setup() throws Exception {
+        wb = createWorkbasketWithPermission();
+        Attachment attachment = createAttachment().build();
+        taskSummary1 =
+            taskInWorkbasket(wb).attachments(attachment).buildAndStoreAsSummary(taskService);
+        taskSummary2 =
+            taskInWorkbasket(wb)
+                .attachments(attachment.copy(), attachment.copy())
+                .buildAndStoreAsSummary(taskService);
+        taskSummary3 = taskInWorkbasket(wb).buildAndStoreAsSummary(taskService);
+        taskSummary4 = taskInWorkbasket(wb).buildAndStoreAsSummary(taskService);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ReturnTasksThatDontHaveAttachment() {
+        List<TaskSummary> list =
+            taskService.createTaskQuery().workbasketIdIn(wb.getId()).withoutAttachment().list();
+
+        assertThat(list).containsExactlyInAnyOrder(taskSummary3, taskSummary4);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @TestFactory
+      Stream<DynamicTest>
+          should_ThrowException_For_InvalidParameterCombinationIncludingWithoutAttachment() {
```

Comment:
I might have an idea to improve this testfactory. Let me show this to you during a video call. Maybe you like it aswell :)